src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -716,27 +716,29 @@ void snc_quota_manager::maybe_set_quota(
 
 void snc_quota_manager::adjust_quota(
   const ingress_egress_state<quota_t>& delta) noexcept {
-    _shard_quota.in.set_quota(_shard_quota.in.quota() + delta.in);
-    _shard_quota.eg.set_quota(_shard_quota.eg.quota() + delta.eg);
-    vlog(klog.trace, "qm - Adjust quota: {} -> {}", delta, _shard_quota);
-}
-
-snc_quota_manager::quota_t
-snc_quota_manager::dispense_quota_amount::get_delta_or_0() const noexcept {
-    if (amount_is == amount_t::delta) {
-        return amount;
+    if (_shard_quota.in.quota() <= -delta.in) {
+        vlog(
+          klog.error,
+          "qm - Adjust quota delta.in {} would make quota {} negative, capping "
+          "at 1",
+          delta.in,
+          _shard_quota.in.quota());
+        _shard_quota.in.set_quota(1);
```

Comment:
q: Whats the implication of this? That the quota is set to min?

Reply:
The quota set to min instead of `0` or a neative value. The implication is that the overall node-wide quota will be bigger than expected. If happens, the workaround is to disable node-wide quota in cluster config and enable it back again. That actually must not happen, and the branch condition is missing an `unlikely()` on it.